### ISSUE TITLE
Markdown: Incorrect parsing of interpolated Julia expressions

### ISSUE THREAD

**Issue by yurivish** [JuliaLang/julia#25992]
```julia
julia> "$(1 + 1)$(1 + 1)"
"22"

julia> Markdown.parse(raw"$(1 + 1) $(1 + 1)")
:(1 + 1)
  :(1 + 1)

julia> Markdown.parse(raw"$(1 + 1)$(1 + 1)")
(1 + 1)

  (1 + 1)

julia> versioninfo()
Julia Version 0.7.0-DEV.3704
Commit f37d8d7a60 (2018-02-03 03:17 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin16.7.0)
  CPU: Intel(R) Core(TM) i7-3615QM CPU @ 2.30GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-3.9.1 (ORCJIT, ivybridge)
Environment:
  JULIA_EDITOR = subl
```

Two expressions separated by a space parse appropriately as two `Expr`s.

Two expressions _not_ separated by a space parse incorrectly (as far as I can tell) as LaTeX. 

```julia
julia> typeof(first(Markdown.parse(raw"$(1 + 1)$(1 + 1)").content))
Markdown.LaTeX
```

**Comment by yurivish** [comment]
Would changing the parsing here constitute a breaking change?

**Comment by JeffBezanson** [comment]
I agree the markdown parser should propagate parse errors in interpolated expressions rather than silently turning them into strings.

**Comment by fonsp** [comment]
There are actually _four_ different results based on spacing:
```julia
julia> Markdown.parse(raw"$(1 + 1) $(1 + 1)")
:(1 + 1)

  :(1 + 1)

julia> Markdown.parse(raw"$(1 + 1)$(1 + 1)")
(1 + 1)

  (1 + 1)

julia> Markdown.parse(raw" $(1 + 1) $(1 + 1)")
  :(1 + 1) :(1 + 1)

julia> Markdown.parse(raw" $(1 + 1)$(1 + 1)")
  (1 + 1)(1 + 1)
```

**Comment by jlewis91** [comment]
@fonsp In all of these cases, the normal string parsing returns 'intuitive' output:

```
julia> "$(1 + 1) $(1 + 1)"
"2 2"

julia> "$(1 + 1)$(1 + 1)"
"22"

julia> " $(1 + 1) $(1 + 1)"
" 2 2"

julia> " $(1 + 1)$(1 + 1)"
" 22"
```

This 'feels' like a precedence issue, one which would be ameliorated if:

1. Markdown strings are parsed like strings, interpolation takes place greedily
2. Markdown parsing happens on the result

What are your thoughts?

**Comment by fonsp** [comment]
(I think that neither Jeff nor me are the person to ask here, the tag felt a bit spammy, but) perhaps you could take a look at the markdown tests to understand the full context. Markdown interpolation is difficult, and different from string interpolation because:
- `$` is also the latex delimiter, in which case it should not interpolate
- even when interpolation occurs, it is not string interpolation, it is lazy HTML interpolation

About the third suggestion, have a look at similar discussions on https://github.com/JuliaLang/julia/issues?q=is%3Aissue+label%3Amarkdown+

For question about this, I think that the Julia slack would be the best place.

**Comment by jlewis91** [comment]
Thanks for the friendly reply and the reprex which greatly clarifies that there are two distinct processes going on here:

1. Parsing string -> Julia obj representation
2. Julia obj -> string

In the case of interpolation and LaTeX conflicts, the question seems to be what interpretation gets priority in ambiguous cases. Given that there is an alternative LaTeX syntax, but no alternative to the interpolation syntax, my above (very crudely sketched) suggestion is to greedily interpret `$(` as a symbol / function call wherever possible and only after those options are exhausted to parse as LaTeX.

I also agree that Markdown parsing in general (given that Julia allows interpolation and LaTeX blocks) is a true challenge (also reflected in the many indeterminate and unresolved issues on the subject). One first step might be refactoring the unit tests so that they offer a more concise picture of what is going on (in terms of status quo and edge cases); currently it seems like the unit tests have developed organically and aren’t as internally consistent as you might want if in a following step one aimed to resolve some of these longstanding parsing ‘quirks’. I’ll take your suggestion and post to Julia slack, see if there would be support for me attempting a refactor of the tests.

**Comment by gxono** [comment]
Ran into another variant of this same `$` ambiguity, noting it here since it's triggered differently from the adjacent-interpolations case above.

An escaped `\$` that appears *after* a real interpolation `$(...)` in the same non-standard string literal breaks that earlier interpolation, even though the two don't visually overlap or sit next to each other:

```julia
using Markdown
x = 42
md"""Valor: $(x) y después \$100 fin."""
```

Expected: `Valor: 42 y después $100 fin.`
Actual: `Valor: $(x) y después \$100 fin.` (neither part gets processed — the interpolation is left as literal, unevaluated text)

Reordering so the escaped `\$` comes *before* the interpolation instead works fine:

```julia
md"""Primero \$100 y ahora $(x) fin."""
```
renders correctly as `Primero $100 y ahora 42 fin.`

So it's not only about two interpolations being adjacent (the original report) — an escaped dollar sign appearing anywhere *after* a legitimate interpolation within the same literal seems to corrupt that earlier interpolation too. Same underlying `$` double-duty (interpolation trigger vs. LaTeX/escape character) as described above, just a different trigger condition. Ran into this in a Pluto.jl notebook mixing computed values with literal currency amounts (`\$100`) in the same markdown cell.
